subgraphs/burrow/src/handlers/actions.ts

```diff
@@ -0,0 +1,589 @@
+  deposit.amount = BigInt.fromString(amount);
+  deposit.asset = token.id;
+  deposit.market = market.id;
+  deposit.position = position.id;
+
+  deposit.amountUSD = deposit.amount
+    .divDecimal(
+      BigInt.fromI32(10)
+        .pow((token.decimals + token.extraDecimals) as u8)
+        .toBigDecimal()
+    )
+    .times(token.lastPriceUSD!);
+
+  // usage metrics
+  if (position._last_active_timestamp.lt(usageDailySnapshot.timestamp)) {
+    usageDailySnapshot.dailyActiveDepositors += 1;
+  }
+  if (account._last_active_timestamp.lt(usageDailySnapshot.timestamp)) {
+    usageDailySnapshot.dailyActiveUsers += 1;
+    usageHourlySnapshot.hourlyActiveUsers += 1;
+  }
+  usageDailySnapshot.dailyDepositCount += 1;
+  usageDailySnapshot.dailyTransactionCount += 1;
+  usageHourlySnapshot.hourlyDepositCount += 1;
+  usageHourlySnapshot.hourlyTransactionCount += 1;
+
+  position.depositCount += 1;
+  position.balance = position.balance.plus(deposit.amount);
+  position._last_active_timestamp = BigInt.fromU64(
+    receipt.block.header.timestampNanosec / 1000000000
+  );
+
+  // update account
+  if (account.depositCount == 0) {
+    protocol.cumulativeUniqueDepositors += 1;
+    protocol.cumulativeUniqueUsers += 1;
+  }
+  account.depositCount += 1;
+  account._last_active_timestamp = BigInt.fromU64(
+    receipt.block.header.timestampNanosec / 1000000000
+  );
+
+  // deposit amount
+  market.outputTokenSupply = market.outputTokenSupply.plus(
+    amount_to_shares(
+      BI_BD(deposit.amount),
+      market.outputTokenSupply,
+      market._totalDeposited
+    )
+  );
+  market._totalDeposited = market._totalDeposited.plus(BI_BD(deposit.amount));
+
+  // historical
+  market._totalDepositedHistory = market._totalDepositedHistory.plus(
+    deposit.amount
+  );
+  market.cumulativeDepositUSD = market.cumulativeDepositUSD.plus(
+    deposit.amountUSD
+  );
+
+  // snapshot
+  dailySnapshot.dailyDepositUSD = dailySnapshot.dailyDepositUSD.plus(
+    deposit.amountUSD
+  );
+  hourlySnapshot.hourlyDepositUSD = hourlySnapshot.hourlyDepositUSD.plus(
+    deposit.amountUSD
+  );
+  financialDailySnapshot.dailyDepositUSD =
+    financialDailySnapshot.dailyDepositUSD.plus(deposit.amountUSD);
+
+  const positionSnapshot = getOrCreatePositionSnapshot(position, receipt);
+  positionSnapshot.logIndex = logIndex as i32;
+
+  updateMarket(
+    market,
+    dailySnapshot,
+    hourlySnapshot,
+    financialDailySnapshot,
+    receipt
+  );
+
+  positionSnapshot.save();
+  financialDailySnapshot.save();
+  usageDailySnapshot.save();
+  usageHourlySnapshot.save();
+  dailySnapshot.save();
+  hourlySnapshot.save();
+  market.save();
+  account.save();
+  deposit.save();
+  position.save();
+  protocol.save();
+
+  updateProtocol();
+}
+
+/**
+ * @dev Handles deposits made to reserve pool
+ * @notice Admin/Owner can deposit to reserve
+ * @param event args {amount, account_id, token_id}
+ */
+export function handleDepositToReserve(event: EventData): void {
```

Comment:
would this be considered protocol side revenue?

Reply:
no